charts/seed-operatingsystemconfig/original/templates/kubelet/_kubelet-monitor.service

```diff
@@ -13,8 +13,10 @@
     EnvironmentFile=/etc/environment
     {{- if semverCompare "< 1.17" .Values.kubernetes.version }}
     ExecStartPre=/usr/bin/docker run --rm -v /opt/bin:/opt/bin:rw {{ required "images.hyperkube is required" .Values.images.hyperkube }} /bin/sh -c "cp /usr/local/bin/kubectl /opt/bin"
-    {{- else }}
+    {{- else if semverCompare "< 1.19" .Values.kubernetes.version }}
     ExecStartPre=/usr/bin/docker run --rm -v /opt/bin:/opt/bin:rw --entrypoint /bin/sh {{ required "images.hyperkube is required" .Values.images.hyperkube }} -c "cp /usr/local/bin/kubectl /opt/bin"
+    {{- else }}
+    ExecStartPre=sh -c "ID=\"$(/usr/bin/docker run --rm -d -v /opt/bin:/opt/bin:rw {{ required "images.hyperkube is required" .Values.images.hyperkube }})\"; /usr/bin/docker cp \"$ID\":/kubectl /opt/bin; docker stop \"$ID\"; chmod +x /opt/bin/kubectl"
```

Comment:
Any specific reason, why we are using two different strategies to copy the binary for different k8s versions now (`cp` in container / `docker cp`)?

Reply:
The gardener/hyperkube image is based on `scratch` and does not have anything other than the binaries inside, so especially it doesn't have a shell/bash. Hence, we have to use `docker cp` now.

Reply:
Got it, makes sense.
Can we do it always via `docker cp` then, to keep it consistent?

Reply:
Let's keep it like that, please. I tried to harmonize it, but it didn't work well. It's probably cleaner and easier to maintain to have the three cases for the three different hyperkube images we use.

Reply:
Ok, fair enough.